### PR TITLE
Fix the Naptr records serialization issue

### PR DESCRIPTION
**Pull request by WhiteHorseV**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/Models/DnsNaptrData.Serialization.cs [hunk 1]
```diff
@@ -82,7 +82,7 @@ void IJsonModel<DnsNaptrRecordData>.Write(Utf8JsonWriter writer, ModelReaderWrit
             }
             if (Optional.IsCollectionDefined(DnsNaptrRecords))
             {
-                writer.WritePropertyName("NaptrRecords"u8);
+                writer.WritePropertyName("NAPTRRecords"u8);
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/Models/DnsNaptrData.Serialization.cs:85]: Please add test for this change

Comment (WhiteHorseV) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/Models/DnsNaptrData.Serialization.cs:85]: Thanks @ArthurMa1978 Added the tests.
I was not able to add the tests as we needed some server side changes deployed.

**Review by djr-jsr — APPROVED** [review]

**Review by archerzz — DISMISSED** [review]
Need to update test recording. See the error logs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4265737&view=logs&j=19f517c9-ad26-5827-ca69-20f64049c15d&t=64a812af-f087-5e7b-c041-871b1ece27d6&l=1495

**Comment by WhiteHorseV** [comment]
> Need to update test recording. See the error logs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4265737&view=logs&j=19f517c9-ad26-5827-ca69-20f64049c15d&t=64a812af-f087-5e7b-c041-871b1ece27d6&l=1495

Updated the assets repo and updated the assets.json file.

**Review by archerzz — APPROVED** [review]

**Pull request merged**